projects/plugins/jetpack/modules/sharedaddy/sharing.php

```diff
@@ -723,6 +723,10 @@ public function sharing_block_display() {
 			new Share_Email( 'email', array() ),
 			new Share_Reddit( 'reddit', array() ),
 		);
+
+		// Hide the link to Jetpack Sharing settings if no Jetpack Settings found in submenu list
+		global $submenu;
+		$show_jetpack_admin_settings_link = isset( $submenu['jetpack'][1][2] ) && $submenu['jetpack'][1][2] === 'jetpack#/settings';
```

Comment:
[nit] I wonder if we could iterate `$submenu['jetpack']` looking for a match to make the check more robust.